charts/redpanda/templates/statefulset.yaml

```diff
@@ -215,83 +215,38 @@ spec:
           startupProbe:
             exec:
               command:
                 - /bin/sh
                 - -c
-                {{- if (include "admin-internal-tls-enabled" . |fromJson).bool }}
-                - >
-                  curl https://localhost:{{ .Values.listeners.admin.port }}/v1/cluster/health_overview
-                  -svk --cacert /etc/tls/certs/{{ .Values.listeners.admin.tls.cert }}/ca.crt |
-                  awk '{
-                    id = $0; gsub(/.*"controller_id": /, "", id); gsub(/,.*/, "", id)
-                    nd_str = $0; gsub(/.*"nodes_down": \[/, "", nd_str); gsub(/\].*/, "", nd_str)
-                    FS=","
-                    split(nd_str, nd_list)
-                    for (i in nd_list) nodes_down[nd_list[i]]=""
-                    exit (id in nodes_down)
-                  }'
-                {{- else }}
-                - >
-                  curl -sv http://localhost:{{ .Values.listeners.admin.port }}/v1/cluster/health_overview |
-                  awk '{
-                    id = $0; gsub(/.*"controller_id": /, "", id); gsub(/,.*/, "", id)
-                    nd_str = $0; gsub(/.*"nodes_down": \[/, "", nd_str); gsub(/\].*/, "", nd_str)
-                    FS=","
-                    split(nd_str, nd_list)
-                    for (i in nd_list) nodes_down[nd_list[i]]=""
-                    exit (id in nodes_down)
-                  }'
-                {{- end }}
+                - curl --silent --fail -k {{ include "admin-tls-curl-flags" . }} "{{ include "admin-http-protocol" . }}://localhost:{{ .Values.listeners.admin.port }}/v1/node_config" | grep -o '"node_id":[0-9]\+'
             initialDelaySeconds: {{ .Values.statefulset.startupProbe.initialDelaySeconds }}
             failureThreshold: {{ .Values.statefulset.startupProbe.failureThreshold }}
             periodSeconds: {{ .Values.statefulset.startupProbe.periodSeconds }}
+          # the livenessProbe just checks to see that the admin api is listening and returning 200s.
           livenessProbe:
             exec:
               command:
                 - /bin/sh
                 - -c
-                {{- if (include "admin-internal-tls-enabled" . |fromJson).bool }}
-                - >
-                  curl https://localhost:{{ .Values.listeners.admin.port }}/v1/cluster/health_overview
-                  -svk --cacert /etc/tls/certs/{{ .Values.listeners.admin.tls.cert }}/ca.crt
-                {{- else }}
-                - >
-                  curl -sv http://localhost:{{ .Values.listeners.admin.port }}/v1/cluster/health_overview
-                {{- end }}
+                - curl --silent --fail -k {{ include "admin-tls-curl-flags" . }} "{{ include "admin-http-protocol" . }}://localhost:{{ .Values.listeners.admin.port }}/v1/node_config"
```

Comment:
If you use `/v1/ready`, that's a bit cheaper for the server than serializing the node config each time.

You don't necessarily have to read the body of the response ("ready" or "booting"), if you just want a simple endpoint to check if the admin API is up.

Reply:
Thanks!